src/usr/share/gpgmailer/gpgmailer.py

```diff
@@ -25,74 +25,90 @@
         self.logger = logging.getLogger('GpgMailer')
         self.logger.info('Initializing gpgmailer module.')
-        self.new_expiration_warning_messages = False
 
         self.config = config
         self.gpgkeyring = gpgkeyring
         self.gpgkeyverifier = gpgkeyverifier
-        self.gpgmailbuilder = gpgmailbuilder.GpgMailBuilder(self.gpgkeyring, self.config['main_loop_duration'])
+        self.gpgmailbuilder = gpgmailbuilder.GpgMailBuilder(self.gpgkeyring,
+            self.config['main_loop_duration'])
         self.mailsender = mailsender.MailSender(self.config)
 
         self.outbox_path = os.path.join(self.config['watch_dir'], 'outbox')
 
-        # Set expiration_warning_message so that it is not detected as changed immediately
-        #   after gpgmailer-daemon sends an email.
+        self.new_expiration_warning_messages = False
+        # Set this here so that the string equality check in _update_expiration_warning_message
+        #   evaluates to equal.
         self.expiration_warning_message = gpgkeyverifier.get_expiration_warning_message(time.time())
 
         self.logger.info('Done initializing gpgmailer module.')
 
-    # Gpgmailer's main loop. Reads the watch directory and then calls other modules
-    #   to build and send email.
+
+    # GpgMailer's main program loop. Reads the watch directory and then calls other modules
+    #   to build and send e-mail. Also sends warnings about GPG key expirations.
     def start_monitoring(self):
         try:
             while True:
                 loop_start_time = time.time()
 
-                self.recipients = self.gpgkeyverifier.get_valid_recipients(loop_start_time)
-                self.keys = self.gpgkeyverifier.get_valid_keys(loop_start_time)
+                self.valid_recipient_emails =
+                    self.gpgkeyverifier.get_valid_recipient_emails(loop_start_time)
+                self.valid_key_fingerprints =
+                    self.gpgkeyverifier.get_valid_key_fingerprints(loop_start_time)
 
-                new_expiration_warning_message = self.gpgkeyverifier.get_expiration_warning_message(loop_start_time)
+                new_expiration_warning_message =
+                    self.gpgkeyverifier.get_expiration_warning_message(loop_start_time)
                 self._update_expiration_warning_message(new_expiration_warning_message)
                 self._send_email_if_new_expiration_warning_messages()
 
                 for file_name in self._get_file_list():
-                    self.logger.info("Found queued email in file %s." % file_name)
+                    self.logger.info("Found queued e-mail in file %s." % file_name)
                     message_dict = self._read_message_file(file_name)
 
 
                     # Set default subject if the queued message does not have one.
-                    if message_dict['subject'] == None:
+                    if message_dict['subject'] is not None:
```

Comment:
This condition is the opposite of the original. It's going to leave None subject values alone and overwrite defined subject values.